src/auth/digest/Config.cc

```diff
@@ -606,79 +544,6 @@ authenticateDigestStats(StoreEntry * sentry)
         digestauthenticators->packStatsInto(sentry, "Digest Authenticator Statistics");
 }
 
-/* NonceUserUnlink: remove the reference to auth_user and unlink the node from the list */
-
-static void
-authDigestNonceUserUnlink(digest_nonce_h * nonce)
-{
-    Auth::Digest::User *digest_user;
-    dlink_node *link, *tmplink;
-
-    if (!nonce)
-        return;
-
-    if (!nonce->user)
-        return;
-
-    digest_user = nonce->user;
-
-    /* unlink from the user list. Yes we're crossing structures but this is the only
-     * time this code is needed
-     */
-    link = digest_user->nonces.head;
-
-    while (link) {
-        tmplink = link;
-        link = link->next;
-
-        if (tmplink->data == nonce) {
-            dlinkDelete(tmplink, &digest_user->nonces);
-            authDigestNonceUnlink(static_cast < digest_nonce_h * >(tmplink->data));
-            delete tmplink;
-            link = nullptr;
-        }
-    }
-
-    /* this reference to user was not locked because freeeing the user frees
-     * the nonce too.
-     */
-    nonce->user = nullptr;
-}
-
-/* authDigesteserLinkNonce: add a nonce to a given user's struct */
-void
-authDigestUserLinkNonce(Auth::Digest::User * user, digest_nonce_h * nonce)
-{
-    dlink_node *node;
-
-    if (!user || !nonce || !nonce->user)
```

Comment:
FTR; the moved code (now in User.cc) drops the ```|| !nonce->user``` clause. This function was often being called for freshly generated nonce objects whose ```user``` member is a nullptr ... apparently intending this function to set that ```user``` member. But this condition prevents the remainder of this function ever working as intended and setting the ```user``` member.